core/algorithms/onpolicy_sync/storage.py

```diff
@@ -232,11 +239,13 @@ def sampler_select(self, keep_list: Sequence[int]):
         self.memory = self.memory.sampler_select(keep_list)
         self.actions = self.actions[:, keep_list]
         self.prev_actions = self.prev_actions[:, keep_list]
-        self.action_log_probs = self.action_log_probs[:, keep_list]
-        self.value_preds = self.value_preds[:, keep_list]
-        self.rewards = self.rewards[:, keep_list]
         self.masks = self.masks[:, keep_list]
-        self.returns = self.returns[:, keep_list]
+
+        if self.rewards is not None:
+            self.action_log_probs = self.action_log_probs[:, keep_list]
+            self.value_preds = self.value_preds[:, keep_list]
+            self.rewards = self.rewards[:, keep_list]
+            self.returns = self.returns[:, keep_list]
 
     def narrow(self):
         assert len(self.unnarrow_data) == 0, "attempting to narrow narrowed rollouts"
```

Comment:
> I know this is a bit beyond the scope of your changes but could you add a comment describing this function and how it's generally used?

I added
"""
This function is used by the training engine (in decentralized distributed settings) to temporarily narrow the
step dimension in the storage. The reverse operation, `unnarrow`, is automatically called by `after_update`.
"""

Reply:
> Also, do we need any check to make sure that `narrow` isn't called twice in a row or will this already cause an error? I'm worried that `self.unnarrow_data` might get corrupted if things get called twice and then `unnarrow` would break.

The first assertion in the function is intended as such a check.